### PR TITLE
[CHK-2438] Update `closePayment` request body to support Checkout Redirect

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

 * Update `closePayment` request body to support Checkout Redirect
 * refactor `closePayment` tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Depends on pagopa/pagopa-infra#1727

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/ClosePaymentRequestBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package it.pagopa.ecommerce.eventdispatcher.services.v2
+
+class ClosePaymentRequestBuilder {}
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/ClosePaymentRequestBuilder.kt:3]: can we remove this class? seems to be unused

File: api-spec/transactions-api.yaml [hunk 2]
```diff
@@ -0,0 +1,1257 @@
+openapi: 3.0.0
```
Comment (pietro-tota) [inline api-spec/transactions-api.yaml:1]: why adding those specs here?

**Review by pietro-tota — DISMISSED** [review]
Looks good to me, just few trivial issues 🚀

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/ClosePaymentRequestBuilder.kt (hunk 1 above)
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/ClosePaymentRequestBuilder.kt]: Fixed in 2e391fb

File: api-spec/transactions-api.yaml (hunk 2 above)
Comment (giovanniberti) [reply api-spec/transactions-api.yaml]: Fixed in 2e391fb

**Review by pietro-tota — APPROVED** [review]

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt [hunk 3]
```diff
@@ -183,125 +188,169 @@ class NodeService(
...
+              }
+            paymentMethodName = authCompleted.transactionAuthorizationRequestData.paymentMethodName
+            clientId = authCompleted.transactionActivatedData.clientId.name
+          }
+        user = UserDto().apply { type = UserDto.TypeEnum.GUEST }
+      }
+
+    return when (authRequestedData.type!!) {
+      TransactionGatewayAuthorizationRequestedData.AuthorizationDataType.PGS,
+      TransactionGatewayAuthorizationRequestedData.AuthorizationDataType.NPG ->
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt:249]: I would add here a comment for NPG case to handle close payment for paypal apm methods (as a reminder for the future) and wallet

Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt:249]: Added in 620ec21

**Review by pietro-tota — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]

**Pull request merged**